adding client tools (#27)
fixie-ai/ultradox: docs/src/content/docs/tools.mdx

- `description` (string): A clear, concise description of what the tool does.
- `dynamicParameters` (array, optional): List of parameters that can be set by the AI model when using the tool. Each parameter is an object containining:
  - `name` (string): The name of the parameter.

>   - `location` (string): Where the parameter is used ("PARAMETER_LOCATION_UNSPECIFIED", "PARAMETER_LOCATION_QUERY", "PARAMETER_LOCATION_PATH", "PARAMETER_LOCATION_HEADER", "PARAMETER_LOCATION_BODY").

Reviewer: ```suggestion
  - `location` (string): Where the parameter is used ("PARAMETER_LOCATION_QUERY", "PARAMETER_LOCATION_PATH", "PARAMETER_LOCATION_HEADER", "PARAMETER_LOCATION_BODY").
```

(unspecified isn't actually valid. It's a placeholder for "unset")

Reviewer: Ya, makes sense.  FWIW all our enums will have an UNSPECIFIED value that's equivalent to not setting the field (and which is thus illegal if the field is required).  It's just a quirk of how OpenAPI gets generated from proto files.

- `automaticParameters` (array, optional): Additional parameters automatically set by the system. Used when the value is not known when the tool is created but that will be known when the tool is called. Example: you want to use the unique `call_id` from ultravox as a key in your backend and you have the tool include `call_id` in the request body when your tool's API is called.
- `requirements` (object, optional): Any specific requirements for using the tool. Currently this is used for security (e.g. API keys or HTTP Auth).
- `http` (object): Details for invoking the tool via HTTP. For server tools.
  - `baseUrlPattern` (string): The base URL pattern for  the tool, possibly with placeholders for path parameters.
  - `httpMethod` (string): The HTTP method for the tool (e.g., "GET", "POST").

> - `client` (object): Declares the tool as a client tool. Either `http` or `client` can be set for a tool.

Reviewer: ```suggestion
- `client` (object): Declares the tool as a client tool. Exactly one of `http` or `client` must be set for a tool.
```
